### PR TITLE
Restore the batch schema bridge for hybrid CP sub-samples

### PR DESCRIPTION
**Pull request by ilml**
## What does this PR do?

Fixes the missing schema bridge that makes `--hybrid-context-parallel` crash on
its first training step. `HybridCPDataLoaderWrapper.unpack_batch` emits 1-D
sub-sample dicts with token-level keys only (it deliberately strips
`cu_seqlens`/`max_seqlen`), while `get_batch_on_this_tp_rank` under
`is_hybrid_cp` requires 2-D batched tensors plus `cu_seqlens`, `max_seqlen`
and `local_cp_size`. The helper that used to bridge the two schemas was
removed in the get_batch consolidation refactor. Symptom: `IndexError` on
`batch['tokens'].shape[1]` at TP=1, asymmetric-broadcast hang at TP>1. Also
fixes `local_cp_size` being a 0-dim CPU tensor while receiving TP ranks
allocate a `(1,)` CUDA buffer for its broadcast.

## Test

The existing schedule unit test previously fed placeholder `{"id": N}` dicts;
it now feeds realistic 1-D `unpack_batch`-shaped sub-samples and asserts the
full bridged 2-D schema, so re-deleting the bridge fails the test. (The only
prior hybrid-CP coverage hand-built the post-bridge schema, which is why this
was invisible to CI.)

## Validation

End to end on 4x GB200: 8B GPT, 100k-token THD workload, mixed CP1/2/4 groups,
finite losses/grad norms. One of three independent fixes required to make
hybrid CP runnable (siblings: runtime-attention and rerun-iterator PRs);
supersedes the combined draft #6816.

### CODE REVIEW

**Review by Phlip79 — APPROVED** [review]

File: megatron/core/pipeline_parallel/hybrid_cp_schedule.py [hunk 1]
```diff
@@ -543,8 +543,26 @@ def _get_new_data_iterator(sample_id_in_group, group_id):
             partner_cp_size = len(
                 [True for sample_ids in sample_id_groups[group_id] if sub_sample_id in sample_ids]
             )
-            sample["local_cp_size"] = torch.tensor(partner_cp_size, dtype=torch.int32)
-            new_data_iterator = RerunDataIterator(iter([sample]))
+            # Bridge the 1-D sub-sample dict from HybridCPDataLoaderWrapper.unpack_batch
```
Comment (zhongbozhu) [inline megatron/core/pipeline_parallel/hybrid_cp_schedule.py:546]: this code change feels unnecessarily complicated, can you share a bit more on why this is necessary when how did you find it in your testing?

**Review by zhongbozhu — APPROVED** [review]

**Review by guihong-nv — APPROVED** [review]

**Review by jaredcasper — APPROVED** [review]

**Comment by ilml** [comment]
/ok to test [7af0919](https://github.com/NVIDIA/Megatron-LM/pull/6820/commits/7af0919d38e442d1c37ddbd7609830023dc47c46)
